### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by Anastastasia-B**
Updated the dependencies

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [ignore]
+<PROJECT_ROOT>/node_modules/resolve/test/resolver/malformed_package_json/package.json
```
Comment (Anastastasia-B) [inline .flowconfig:2]: Looks like the only workaround currently https://github.com/browserify/resolve/issues/271#issuecomment-1021307415

File: packages/eslint-config-base/__tests__/linter.js [hunk 2]
```diff
@@ -4,19 +4,30 @@ const eslint = require('eslint')
...
+    ...config,
+    rules: {
+      ...config.rules,
+      'no-unused-vars': 'off',
+      'no-constant-condition': 'off',
+      'no-empty': 'off',
+      'class-methods-use-this': 'off',
+      'import/no-unresolved': 'off',
+      'no-unused-expressions': 'off',
+      'import/extensions': 'off',
```
Comment (Anastastasia-B) [inline packages/eslint-config-base/__tests__/linter.js:19]: Disabled some rules (which are not related with what we are testing) for testing, so it is easier to write code examples, without the need to follow all of them, for example no unused variables or no empty block statements

File: packages/stylelint-config/__tests__/rules/function-url-quotes.js [hunk 3]
```diff
@@ -6,14 +6,14 @@ linter.verify({
...
   valid: [
-    ' @document domain(\'http://www.w3.org/\'); ',
+    ' @import url(\'foo.css\'); ',
     ' .class { background-image: url(\'x.jpg\'); } ',
-    ' .class { background-image: asset-url(); } ',
+    ' .class { background-image: url(); } ',
   ],
 
   invalid: [
-    ' @document domain(http://www.w3.org/); ',
```
Comment (Anastastasia-B) [inline packages/stylelint-config/__tests__/rules/function-url-quotes.js:15]: `@document domain` is deprecated

File: packages/stylelint-config/package.json [hunk 4]
```diff
@@ -13,8 +13,8 @@
     "linter"
   ],
   "peerDependencies": {
-    "stylelint": "^13.8",
-    "stylelint-config-standard": "^20"
+    "stylelint": "^15.10.3",
```
Comment (lesha1201) [inline packages/stylelint-config/package.json:16]: It's better to specify it as `^15.10` to support a wider range of versions. Currently, if a user will have `stylelint@15.10.0` in their dependencies, they would end up with two versions of `stylelint`.


```suggestion
    "stylelint": "^15.10",
```

File: packages/stylelint-config/package.json [hunk 5]
```diff
@@ -13,8 +13,8 @@
...
+    "stylelint-config-standard": "^34.0.0"
```
Comment (lesha1201) [inline packages/stylelint-config/package.json:17]:
```suggestion
    "stylelint-config-standard": "^34.0"
```

File: packages/stylelint-config/index.js
Comment (lesha1201) [inline packages/stylelint-config/index.js:1]: This config has some deprecated rules. I think it makes sense to remove them (https://stylelint.io/migration-guide/to-15/#deprecated-stylistic-rules):

- `indentation`
- `string-quotes`

File: packages/eslint-config-base/index.js [hunk 6]
```diff
@@ -1,7 +1,13 @@
...
 
 module.exports = {
-  parser: 'babel-eslint',
+  parser: '@babel/eslint-parser',
+  parserOptions: {
+    requireConfigFile: false,
+    babelOptions: {
+      presets: ['@babel/preset-flow'],
+    },
+  },
```
Comment (lesha1201) [inline packages/eslint-config-base/index.js:10]: As I understand, previously, that config didn't require Flow and wasn't intended for it. I think we shouldn't rely on Flow here.

File: packages/eslint-config-base/package.json [hunk 7]
```diff
@@ -18,7 +18,8 @@
   "peerDependencies": {
     "@babel/eslint-parser": "^7.18.2",
     "eslint": "^8",
-    "eslint-plugin-import": "^2.16.0"
+    "eslint-plugin-import": "^2.28.1",
+    "@babel/preset-flow": "^7.22.15"
```
Comment (lesha1201) [inline packages/eslint-config-base/package.json:22]:
```suggestion
```

File: packages/eslint-config-base/__tests__/empty-lines.js [hunk 8]
```diff
@@ -65,27 +65,29 @@ linter.verify({
...
 
       case 'one':
       case 'two':
-        return true
+        true
+        break
 
       default:
-        return false
+        false
```
Comment (lesha1201) [inline packages/eslint-config-base/__tests__/empty-lines.js:77]: Why did you change this? What's the point of `false;break;`?

File: packages/eslint-config/package.json [hunk 9]
```diff
@@ -15,14 +15,16 @@
   ],
   "dependencies": {
     "eslint-config-airbnb": "^19.0.4",
-    "eslint-config-datarockets-base": "latest"
+    "eslint-config-datarockets-base": "https://gitpkg.now.sh/datarockets/style-guide/packages/eslint-config-base?upgrade-dependencies"
```
Comment (lesha1201) [inline packages/eslint-config/package.json:18]:
```suggestion
    "eslint-config-datarockets-base": "latest"
```

I'm not sure how to tell the old Lerna to use the workspace package instead of pinned version but I don't think we should use `https://gitpkg.now.sh/datarockets/style-guide/packages/eslint-config-base?upgrade-dependencies` in a published version.

File: packages/eslint-config/index.js [hunk 10]
```diff
@@ -6,6 +6,13 @@ module.exports = {
     'datarockets-base',
   ],
 
+  parserOptions: {
+    requireConfigFile: false,
+    babelOptions: {
+      presets: ['@babel/preset-react', '@babel/preset-flow'],
+    },
+  },
```
Comment (lesha1201) [inline packages/eslint-config/index.js:14]:
```suggestion
```

File: packages/eslint-config/package.json [hunk 11]
```diff
@@ -15,14 +15,16 @@
...
   "peerDependencies": {
     "@babel/eslint-parser": "^7.18.2",
     "eslint": "^8",
-    "eslint-plugin-import": "^2.16.0",
+    "eslint-plugin-import": "^2.28.1",
     "eslint-plugin-jsx-a11y": "^6.2.1",
-    "eslint-plugin-react": "^7.12.4"
+    "eslint-plugin-react": "^7.12.4",
+    "@babel/preset-flow": "^7.22.15",
+    "@babel/preset-react": "^7.22.15"
```
Comment (lesha1201) [inline packages/eslint-config/package.json:27]:
```suggestion
```

**Comment by lesha1201** [comment]
The project was completely rewritten so this PR isn't relevant anymore.
